### ISSUE TITLE
Implement Deferred Execution for Chainable Methods (limit, sort, where) with Final get() Method

### ISSUE THREAD

**Issue by roberto-butti** [Hi-Folks/fusionable#2]
## Summary

This change request proposes updating the behavior of chainable methods (e.g., limit, sort, where) in the library to defer execution until a final get() method is called. This approach will enhance flexibility, performance, and predictability by allowing all transformations to be applied in a consistent order only when get is invoked.

## Current Behavior
Currently, each chainable method (such as limit, sort, where) immediately modifies the data array when called. 
With the current implementation:

- Each method (where, sort, and limit) applies its transformation as soon as it’s called.
- The order of calling these methods directly impacts the results, which may lead to unintended behavior and complexity.

## Proposed New Behavior

This change request proposes deferring the application of transformations until a final get() method is called.

- Methods like where, sort, and limit would collect parameters or operations without immediately modifying the data.
- Calling get() would trigger the execution of all collected operations in the order they were called.
- This deferred execution approach ensures transformations are applied predictably and efficiently, with only a single pass through the data.

## Benefits

Switching to deferred execution provides several advantages:

- Predictable and Consistent Execution Order: Users can chain methods without worrying about the order in which transformations are applied, as they will always be applied in the order specified in the chain.
- Performance Improvements: Deferred execution reduces the need for intermediate transformations. For example, multiple where filters can be combined and applied in one pass during get(), minimizing redundant processing.
- Enhanced Readability and Flexibility: The new approach provides a clear structure where transformations are merely configured with each method and executed only once with get(). This makes code easier to read and understand.

## Impact
This change may affect existing code that relies on the current immediate-execution behavior of methods. To mitigate potential issues:

- Documentation Update: Update documentation to explain the deferred execution model and show examples with get().
- Change also the documentation
- No worries about retro compatibility; we are in a super early stage (a kind of preview for collecting feedback), and we are still in 0.0.x .

**Comment by roberto-butti** [comment]
closed by #3
